packages/docusaurus-theme-classic/src/theme/DocSidebar/index.tsx

```diff
@@ -43,31 +47,67 @@ const isActiveSidebarItem = (item, activePath) => {
   return false;
 };
 
+type CollapseMap = Record<string, boolean | undefined>;
```

Comment:
why `| undefined` ? the category is collapsed or is not collapsed, there is no other possible state for a category